src/transformers/trainer.py

```diff
@@ -1170,6 +1186,121 @@ def get_optimizer_cls_and_kwargs(args: TrainingArguments) -> Tuple[Any, Any]:
             optimizer_cls = torch.optim.Adagrad
         elif args.optim == OptimizerNames.RMSPROP:
             optimizer_cls = torch.optim.RMSprop
+        elif args.optim in [
+            OptimizerNames.GALORE_ADAMW,
+            OptimizerNames.GALORE_ADAMW_8BIT,
+            OptimizerNames.GALORE_ADAFACTOR,
+            OptimizerNames.GALORE_ADAMW_LAYERWISE,
+            OptimizerNames.GALORE_ADAMW_8BIT_LAYERWISE,
+            OptimizerNames.GALORE_ADAFACTOR_LAYERWISE,
+        ]:
+            if not is_galore_torch_available():
+                raise ImportError(
+                    "You need to install `galore_torch` in order to use GaLore optimizers"
+                    " install it with `pip install git+https://github.com/jiaweizzhao/GaLore`"
+                )
+
```

Comment:
Since it doesn't support `DDP`, let's add a check here on the distributed type and raise an error for users? (basically check if `self.args.parallel_mode == ParallelMode.DISTRIBUTED`)
```suggestion
            if self.args.parallel_mode == ParallelMode.DISTRIBUTED:
                raise NotImplementedError("GaLore does not support DDP at this time")
```

Reply:
Yes ! note only the layer-wise optimizers do not support DDP, so I updated the logic accordingly